easybuild/tools/modules.py

```diff
@@ -504,11 +504,12 @@ def run_module(self, *args, **kwargs):
         if kwargs.get('return_output', False):
             return stdout + stderr
         else:
-            # the module command was run with an outdated LD_LIBRARY_PATH, which will be adjusted on loading a module
+            # the module command was run with an outdated $LD_LIBRARY_PATH and $LD_PRELOAD,
+            # which will be adjusted on loading a module;
             # this needs to be taken into account when updating the environment via produced output, see below
 
-            # keep track of current LD_LIBRARY_PATH, so we can correct the adjusted LD_LIBRARY_PATH below
-            prev_ld_library_path = os.environ.get('LD_LIBRARY_PATH', '').split(':')[::-1]
+            # keep track of current $LD_LIBRARY_PATH/$LD_PRELOAD, so we can correct the adjusted values below
+            prev_ld_values = dict([(key, os.environ.get(key, '').split(os.pathsep)[::-1]) for key in LD_ENV_VAR_KEYS])
```

Comment:
different environment than the previous time, and these environment variables may be changed in between execution of module commands; this is correct